### PR TITLE
feat: speed up fallback translations plugin

### PR DESCRIPTION
**Pull request by oleksii-leonov**
<details>
  <summary>PR Checklist</summary>

### PR Structure

- [x] This PR has reasonably narrow scope (if not, break it down into smaller
      PRs).
- [x] This PR avoids mixing refactoring changes with feature changes (split into
      two PRs otherwise).
- [x] This PR's title starts is concise and descriptive.

### Thoroughness

- [x] This PR adds tests for the most critical parts of the new functionality or
      fixes.
- [x] I've updated any docs, `.md` files, etc… affected by this change.

</details>

### What

Speed up the fallback translations plugin.
`deep_merge` code extracted from `i18n-js` v3.

### Why

Current versions with `Glob` became too slow on large localization files.
For our use case (20 locales, 20000 keys each), I wait ~20 minutes and kill the process.

`deep_merge` versions from `v3` works instantly. So, I have extracted the `deep_merge` from `v3` into `v4`.

### Known limitations

N/A

### CODE REVIEW

**Comment by fnando** [comment]
This is great. Thank you! I did just a few synthetic tests with no more than a hundred translations and it's okay, but your real use case caught the performance introduced by glob (i'm pretty sure there's an opportunity for optimization there).

File: lib/i18n-js/embed_fallback_translations_plugin.rb [hunk 1]
```diff
@@ -16,33 +44,21 @@ def validate_schema
...
-        buffer[locale] = Glob.new(translations[locale]).tap do |glob|
-          glob << "*"
-        end
-      end
-
-      default_locale = I18n.default_locale
-      default_locale_glob = mapping.delete(default_locale)
-      default_locale_paths = default_locale_glob.paths
+      fallback_locale = I18n.default_locale.to_sym
+      locales_to_fallback = translations.keys - fallback_locale
```
Comment (fnando) [inline lib/i18n-js/embed_fallback_translations_plugin.rb:50]: Shouldn't this be `[fallback_locale]` instead? I'm running this in development and I'm getting an exception.

```suggestion
      locales_to_fallback = translations.keys - [fallback_locale]
```

**Comment by fnando** [comment]
@oleksii-leonov just pushed to your repo with some changes related to uncovered code (just to make sure those mergers were working as expected).

File: lib/i18n-js/embed_fallback_translations_plugin.rb (hunk 1 above)
Comment (oleksii-leonov) [reply lib/i18n-js/embed_fallback_translations_plugin.rb]: Sorry, my bad. It should be `translations.keys - [fallback_locale]` 👍

**Pull request merged**